hermeto/core/resolver.py

```diff
@@ -62,17 +62,50 @@ def resolve_packages(request: Request) -> RequestOutput:
 
 
 def _resolve_packages(request: Request) -> RequestOutput:
-    """Run all requested package managers, return their combined output."""
+    """
+    Run all requested package managers, return their combined output.
+
+    Supports individual experimental flags `x-<name>` alongside `dev-package-managers`.
+    """
     _supported_package_managers = _package_managers
-    requested_types = set(pkg.type for pkg in request.packages)
-    if "dev-package-managers" in request.flags:
-        _supported_package_managers = _package_managers | _dev_package_managers
+    requested_types: set[PackageManagerType] = {pkg.type for pkg in request.packages}
+
+    # handle individual experimental package managers toggles
+    experimental_package_managers: set[PackageManagerType] = {
+        type for type in requested_types if type.startswith("x-")
+    }
+    if experimental_package_managers and "dev-package-managers" in request.flags:
+        raise UnsupportedFeature(
+            "Experimental package manager not supported with --dev-package-managers tag.",
+            solution="Switch to using only x-<pkg> please.",
+        )
+    elif experimental_package_managers:
+        requested_types = {
+            cast(PackageManagerType, t.removeprefix("x-")) if t.startswith("x-") else t
+            for t in requested_types
+        }
+        # Normalize each experimental pkg and enable only those dev managers
+        normalized: set[PackageManagerType] = {
+            cast(PackageManagerType, t.removeprefix("x-")) for t in experimental_package_managers
+        }
+        for pkg in normalized:
+            _supported_package_managers[pkg] = _dev_package_managers[pkg]
+
+        # Replace x-<pkg> in requested_types with the normalized names
+        requested_types = (requested_types - experimental_package_managers) | normalized
+    # handle global dev-package-managers flag
+    elif "dev-package-managers" in request.flags:
+        _supported_package_managers |= _dev_package_managers
+
+    # check for any unsupported
```

Comment:
nitpick: pre-existing, but I'm not sure if `unsupported_types` describe the situation correctly, garbage values are also strictly unsupported, but those will never get up to this point, so using `experimental` in the identifiers may describe the situation in the code better.